src/Components/CoffeeStoreContainer.tsx

```diff
@@ -1,27 +1,41 @@
 import { spinnerService } from '@simply007org/react-spinners';
 import React, { useEffect, useState } from 'react';
-import { translate } from 'react-translate';
 import { Client } from '../Client';
 import { matchesTaxonomy } from '../Utilities/CheckboxFilter';
-import { defaultLanguage, initLanguageCodeObject } from '../Utilities/LanguageCodes';
+import { defaultLanguage, initLanguageCodeObjectWithArray } from '../Utilities/LanguageCodes';
 import CheckboxFilter from './CheckboxFilter';
 import CoffeeStoreListing from './CoffeeStoreListing';
+import { ITaxonomyTerms } from '@kentico/kontent-delivery';
+import { Coffee } from '../Models/coffee';
+import { useIntl } from 'react-intl';
+import { projectModel } from '../Models/_project';
 
-const CoffeeStoreContainer = ({ language, t }) => {
-  const [coffees, setCoffees] = useState(initLanguageCodeObject());
-  const [processings, setProcessings] = useState([]);
-  const [productStatuses, setProductStatuses] = useState([]);
+interface CoffeeStoreContainerProps {
+  language: string
+}
 
-  const [filter, setFilter] = useState({
+interface filterType {
```

Comment:
Still thinking whether to place all interfaces + types ant one place i.r. ~/types.d.ts

Reply:
As I found out *.d.ts are mostly used to type some javascript code e.g. libraries that were written in plain javascript and we as typescript developers want have type checking, because of that I have decided to keep interfaces in the files.